Support Asymptote diagrams

mgajda opened the pull request:
Fixes #55 by adding supported for Asymptote.

LaurentRDC commented:
Thank you for your contribution!

Can you also modify the [continuous integration script](https://github.com/LaurentRDC/pandoc-plot/blob/master/.github/workflows/ci.yml) to install Asymptote? You'll also need to modify the ['integration tests' step](https://github.com/LaurentRDC/pandoc-plot/blob/master/.github/workflows/ci.yml#L155) to test pandoc-plot on `issue55.md`

mgajda commented:
The CI issues are with Python Bokeh.
I could not find an asymptote installation GitHub action, so it is hard to test.

mgajda commented:
It seems that Asymptote works on all platforms.

If you approve this and #62, I may check how much to add to MacOS tests that would still work.

mgajda commented:
Please merge!

LaurentRDC commented on tests/issue55.md:
````diff
@@ -0,0 +1,74 @@
+
+# Example Asymptote plot
+
+```{.asy}
````
The default PNG format isn't supported; please change this to:

````{markdown
```{.asy format=pdf}
````

mgajda replied:
Found reason for PNG issues, and added support for default PNG format.
However, I would recommend using PDF or EPS as default output formats, since they are vector formats.

mgajda commented:
1. I have added testing of all output formats for all available toolkits.
2. I have added declaring output format for Asymptote: PDF, EPS, and PNG work.
3. Unfortunately SVG, JPG output do not work in default Debian installation due to stringent ImageMagick security policy.

mgajda commented:
Previous round of tests was success with Asymptote on Mac:
https://github.com/LaurentRDC/pandoc-plot/actions/runs/6174543606/job/16759375009?pr=61#step:20:175
Unfortunately the executables are undetected on Ubuntu and Windows. For unknown reasons.

mgajda commented:
Since `asymptote` tests work now, I suggest we merge this, and I continue fighting CI in a separate PR.

LaurentRDC commented:
I'm not merging until all the CI tests pass. 

I committed earlier today and [the tests all passed](https://github.com/LaurentRDC/pandoc-plot/actions/runs/6172684265). Therefore, the source of breakage is in this PR.

Looks like some versions toolkits (e.g.MATLAB and ggplot2) installed in Github CI specifically don't support all save formats. E.g. I can save SVGs in MATLAB on my laptop, but apparently not in GitHub CI. Therefore, [your update to the tests](https://github.com/LaurentRDC/pandoc-plot/pull/61/files#diff-94a2066a7978fb8d2ad6e399d4cc6db25eff7ec3b14305026cb067bada7f79c9R129) is too optimistic.

If you undo that change, I think the CI tests will all pass

mgajda commented:
Thanks for feedback!

I agree that I should undo CI changes, and ensure all tests pass.

LaurentRDC commented on .github/workflows/ci.yml:
```diff
@@ -120,13 +124,22 @@ jobs:
         dot -c
 
         choco install --yes --no-progress gnuplot
+        #refreshenv
```
Can you remove commented-out commands?

LaurentRDC commented:
This is looking really good. I've made one last comment to clean up the PR a bit and remove CI debugging commands.

Once that's done, I'll merge your PR.

LaurentRDC commented:
I'll be cleaning up the PR directly in the master branch.

Thank you for your contribution!
